tests/activity_test.go

```diff
@@ -338,6 +339,7 @@ func (s *functionalSuite) TestActivityRetry() {
 		})
 	}
 
+	// let poller[1] to process tasks and assert that it completed successfully
```

Comment:
by `poller[1]` do you mean the variable `poller` as opposed to `poller2`? if so, I think it would be more clear to rename `poller` --> `poller1` and refer to the new variable name in the comment, because `poller[1]` makes me think there's an array somewhere.

as someone new who doesn't know what `PollAndProcessWorkflowTask()` does yet, it would be helpful for this comment to read: `assert that workflow tasks completed successfully`, but I think it would actually be even more helpful if that was commented on the top of the `PollAndProcessWorkflowTask` fn itself (in `taskpoller.go`)

Reply:
actually on further reading I'm seeing that the behavior of `PollAndProcessTask` (workflow or activity) depends on the task handler passed to the poller, so I think `assert that workflow tasks completed successfully` is not an accurate comment for that function. Instead it would have to be something like `// polls workflow tasks and handles them as described in poller's WorkflowTaskHandler`

Reply:
yes, by poller[1] I mean poller. With this pull request I wanted to introduce as minimum change as possible and at the same time remind that there are more then one poller. 
When I think that I need 2 values which do not have 2 distinct roles (meaning I cant find 2 distinct meaningful names) I usually use an array. 
The author of the test created 2 pollers and I am not changing the test. So, I emphasize that it was done by a specific poller and not any poller. I can rephrase my comment as "assert that all the pending tasks were processed successfully by poller[1]".